cmake/CMakeLists.txt

```diff
@@ -0,0 +1,166 @@
+project(usb-1.0 C)
+cmake_minimum_required(VERSION 3.10)
+
+# This is because CMake on mingw can incorrectly use response files for includes
+# instead of passing them directly via -I<dir>
+# Explicitly disable use of response files
+set(CMAKE_C_USE_RESPONSE_FILE_FOR_INCLUDES   OFF)
+
+include(CheckIncludeFiles)
+include(CheckFunctionExists)
+include(CheckSymbolExists)
+include(CheckStructHasMember)
+include(CheckCCompilerFlag)
+include(CTest)
+
+check_function_exists(clock_gettime             HAVE_CLOCK_GETTIME)
+check_function_exists(pthread_condattr_setclock HAVE_PTHREAD_CONDATTR_SETCLOCK)
+check_function_exists(pthread_setname_np        HAVE_PTHREAD_SETNAME_NP)
+check_function_exists(pthread_threadid_np       HAVE_PTHREAD_THREADID_NP)
+check_function_exists(eventfd                   HAVE_EVENTFD)
+check_function_exists(pipe2                     HAVE_PIPE2)
+check_function_exists(syslog                    HAVE_SYSLOG)
+
+check_include_files(asm/types.h      HAVE_ASM_TYPES_H) 
+check_include_files(sys/eventfd.h    HAVE_EVENTFD) 
+check_include_files(dlfcn.h          HAVE_DLFCN_H)
```

Comment:
A bunch of these are totally unused.

Reply:
Indeed there are no issues removing them. Tested with both VS2022 and MSYS2 mingw64.

Reply:
I believe those are added to fully mimic the autotools-generated version of `config.h`